packages/viz/src/lib/Layer.ts

```diff
@@ -1,107 +1,107 @@
-import { Credentials, defaultCredentials } from '@carto/toolkit-core';
-import { MapInstance, MapOptions, Maps } from '@carto/toolkit-maps';
 import { MVTLayer } from '@deck.gl/geo-layers';
 
-import Source from './Source';
-import {defaultStyles, Style} from './style';
+import { CARTOSource } from './sources/CARTOSource';
+import { Source } from './sources/Source';
 
-const defaultMapOptions: MapOptions = {
-  vector_extent: 2048,
-  vector_simplify_extent: 2048,
-  metadata: {
-    geometryType: true
-  }
-};
+import {defaultStyles, Style} from './style';
 
 export class Layer {
-  private _mapsClientInstance: Maps;
-  private _credentials: Credentials;
 
-  private _layerSource: Source;
-  private _layerStyles: Style;
-  private _layerOptions: MapOptions;
-  private _layerInstantiation: Promise<MapInstance>;
+  private _source: Source;
+  private _styles: Style;
 
   private _deckInstance: any;
-  private _mvtLayerInstance: MVTLayer<string> | undefined;
-
-  constructor(source: string, styles = {}, options: LayerOptions = {}) {
-    const { mapOptions = {}, credentials = defaultCredentials } = options;
-
-    this._credentials = credentials;
-    this._mapsClientInstance = new Maps(this._credentials);
+  private _deckLayer: any;
 
-    this._layerSource = new Source(source);
-    this._layerStyles = new Style(styles);
-
-    this._layerOptions = Object.assign({}, defaultMapOptions, mapOptions);
-    this._layerInstantiation = this._mapsClientInstance.instantiateMapFrom(
-      buildInstantiationOptions({ mapOptions: this._layerOptions, mapSource: this._layerSource })
-    );
+  constructor(source: string | CARTOSource, styles = {}) {
```

Comment:
We have lost `LayerOptions` param here, so we broke the use of credentials, as I noticed from Layer.spec.ts.  Did you ommit that on purpouse for any particular reason? If not we should get them back, and inject it to Source ctor.